src/decoder/mod.rs

```diff
@@ -1505,10 +1546,9 @@ impl<R: Read + Seek> Decoder<R> {
 
     /// Decodes the entire image and return it as a Vector
     pub fn read_image(&mut self) -> TiffResult<DecodingResult> {
-        let result = match (self.image().chunk_type, self.image().compression_method) {
-            (_, CompressionMethod::ModernJPEG) => self.read_jpeg()?,
-            (ChunkType::Strip, _) => self.read_stripped_image()?,
-            (ChunkType::Tile, _) => self.read_tiled_image()?,
+        let result = match self.image().chunk_type {
```

Comment:
nit: instead of having a result variable, we could just wrap the entire match statement in `Ok(...)`